snippets/base/admin/adminmodels.py

```diff
@@ -261,9 +260,25 @@ def get_form(self, request, obj=None, **kwargs):
 
     @transaction.atomic
     def make_published(self, request, queryset):
-        for snippet in queryset:
-            snippet.status = STATUS_CHOICES['Published']
-            snippet.save()
+        clean_queryset = queryset.exclude(status=models.STATUS_CHOICES['Published'])
+        no_snippets = clean_queryset.count()
+        no_already_published_snippets = queryset.count() - no_snippets
+
+        snippets = []
+        with transaction.atomic():
```

Comment:
This makes the `transaction.atomic` decorator on the method redundant (and we wouldn't want the DB to get locked waiting for responses from slack), so I assume you just forgot to remove it?

Reply:
Yes i forgot to remove it here, good catch. It's worth mentioning though that if I was to handle any possible exceptions, this is the way to do it according to the [docs](https://docs.djangoproject.com/en/2.1/topics/db/transactions/) as you shouldn't be handling exception within the atomic block.